golang/internal/mapper/grpc.go

```diff
@@ -375,46 +375,48 @@ func mapVolumeLinks(in []*agent.VolumeLink) []v1.VolumeLink {
 	return volumeLinks
 }
 
-func MapContainerState(in []dockerTypes.Container, prefix string) []*common.ContainerStateItem {
-	list := []*common.ContainerStateItem{}
+func MapContainerState(it *dockerTypes.Container, prefix string) *common.ContainerStateItem {
+	name := ""
+	if len(it.Names) > 0 {
+		name = strings.TrimPrefix(it.Names[0], "/")
+	}
 
-	for i := range in {
-		// TODO (@m8Vago): get rid of it :)
-		it := in[i]
+	if prefix != "" {
+		name = strings.TrimPrefix(name, prefix+"-")
+	}
 
-		name := ""
-		if len(it.Names) > 0 {
-			name = strings.TrimPrefix(it.Names[0], "/")
-		}
+	imageName := strings.Split(it.Image, ":")
 
-		if prefix != "" {
-			name = strings.TrimPrefix(name, prefix+"-")
-		}
+	var imageTag string
 
-		imageName := strings.Split(it.Image, ":")
+	if len(imageName) > 1 {
+		imageTag = imageName[1]
+	} else {
+		imageTag = "latest"
+	}
 
-		var imageTag string
+	return &common.ContainerStateItem{
+		Id: &common.ContainerIdentifier{
+			Prefix: prefix,
+			Name:   name,
+		},
+		Command:   it.Command,
+		CreatedAt: timestamppb.New(time.UnixMilli(it.Created * int64(time.Microsecond)).UTC()),
+		State:     MapDockerStateToCruxContainerState(it.State),
+		Reason:    it.State,
+		Status:    it.Status,
+		Ports:     mapContainerPorts(&it.Ports),
+		ImageName: imageName[0],
+		ImageTag:  imageTag,
+	}
+}
 
-		if len(imageName) > 1 {
-			imageTag = imageName[1]
-		} else {
-			imageTag = "latest"
-		}
+func MapContainerStates(in []dockerTypes.Container, prefix string) []*common.ContainerStateItem {
```

Comment:
`MapContainerStateList` so we're not going to confuse it with `MapContainerState`